opensrp-opd/src/main/java/org/smartregister/opd/OpdLibrary.java

```diff
@@ -193,7 +247,7 @@ public OpdRulesEngineHelper getOpdRulesEngineHelper() {
     }
 
     @NonNull
-    public Event processOpdCheckInForm(@NonNull String eventType, String jsonString, @Nullable Intent data) throws JSONException {
+    public List<Event> processOpdCheckInForm(@NonNull String eventType, String jsonString, @Nullable Intent data) throws JSONException {
```

Comment:
Why change the return type here if only one event is going to be returned?

Reply:
I will need to save the multiple events for diagnosis and treat  https://github.com/OpenSRP/opensrp-client-opd/blob/456236493be9e7cf5155530f62c9179cd56b0561/opensrp-opd/src/main/java/org/smartregister/opd/presenter/BaseOpdRegisterActivityPresenter.java#L94